react-native/react/pinentry/pinentry-render.js

```diff
@@ -1,30 +1,47 @@
 'use strict'
 /* @flow */
 
-import React from 'react'
-import BaseComponent from '../base-component'
-import { TextField, RaisedButton } from 'material-ui'
+import React, { Component } from '../base-react'
+import { TextField, RaisedButton, Checkbox } from 'material-ui'
 
-export default class PinentryRender extends BaseComponent {
+export default class PinentryRender extends Component {
   componentWillMount () {
     this.state = {
       passphrase: ''
     }
+    for (let feature in this.props.payload.features) {
+      if (this.props.payload.features[feature].hasOwnProperty('value')) {
+        this.state[feature] = this.props.payload.features[feature].value
```

Comment:
this seems dangerous, maybe we should make a part of state for these features instead of the root. i could inject passphrase and it'd overwrite passphrase right?